common/workunit/workunit.cpp

```diff
@@ -5389,15 +5556,44 @@ class CDaliWorkUnitFactory : public CWorkUnitFactory, implements IDaliClientShut
         return "Dali";
     }
 
-    virtual CLocalWorkUnit *_createWorkUnit(const char *wuid, ISecManager *secmgr, ISecUser *secuser)
+    virtual CLocalWorkUnit *_createWorkUnit(const char *wuid, IPropertyTree *wuXML, ISecManager *secmgr, ISecUser *secuser)
     {
         StringBuffer wuRoot;
         getXPath(wuRoot, wuid);
         IRemoteConnection *conn;
         conn = sdsManager->connect(wuRoot.str(), session, RTM_LOCK_WRITE|RTM_CREATE_UNIQUE, SDS_LOCK_TIMEOUT);
-        conn->queryRoot()->setProp("@xmlns:xsi", "http://www.w3.org/1999/XMLSchema-instance");
-        conn->queryRoot()->setPropInt("@wuidVersion", WUID_VERSION);
-        conn->queryRoot()->setProp("@totalThorTime", "");
+        if (!wuXML)
+        {
+            conn->queryRoot()->setProp("@xmlns:xsi", "http://www.w3.org/1999/XMLSchema-instance");
+            conn->queryRoot()->setPropInt("@wuidVersion", WUID_VERSION);
+            conn->queryRoot()->setProp("@totalThorTime", "");
+        }
+        else
+        {
+            Owned<IPropertyTree> wuXMLPTree(wuXML);
+
+            IPropertyTree *root = conn->queryRoot();
+            const char *wuidUnique = root->queryName();
+            wuXMLPTree->renameProp("/", wuidUnique);
+
+            Owned<IPropertyTree> progressPTree = pruneBranch(wuXMLPTree, "GraphProgress[1]");
```

Comment:
The [1] shouldn't be necessary, it implies there could be >1 GraphProgress node..